### PR TITLE
Hotfix edges

### PR DESCRIPTION
**Pull request by colin-grapl**
### Which issue does this PR correspond to?
#324

### What changes does this PR make to Grapl? Why?
This makes EdgeT definitions far less confusing.

### How were these changes tested?
I tested locally, and validated the generated schemas in DGraph.

### CODE REVIEW

File: src/python/grapl-model-plugin-deployer/src/grapl_model_plugin_deployer.py [hunk 1]
```diff
@@ -134,7 +134,9 @@ def store_schema(dynamodb, schema: "Schema"):
     table = dynamodb.Table(os.environ["BUCKET_PREFIX"] + "-grapl_schema_table")
     for f_edge, (edge_t, r_edge) in schema.get_edges().items():
         if not (f_edge and r_edge):
+            print(f"missing {f_edge} {r_edge} for {schema.self_type()}")
```
Comment (jgrillo-grapl) [inline src/python/grapl-model-plugin-deployer/src/grapl_model_plugin_deployer.py:137]: `LOGGER`?

File: src/python/grapl-model-plugin-deployer/src/grapl_model_plugin_deployer.py [hunk 2]
```diff
@@ -134,7 +134,9 @@ def store_schema(dynamodb, schema: "Schema"):
...
             continue
+        print(f"{f_edge} {r_edge} {edge_t.rel}")
```
Comment (jgrillo-grapl) [inline src/python/grapl-model-plugin-deployer/src/grapl_model_plugin_deployer.py:139]: same

File: src/python/grapl-model-plugin-deployer/src/grapl_model_plugin_deployer.py [hunk 3]
```diff
@@ -143,6 +145,7 @@ def store_schema(dynamodb, schema: "Schema"):
             }
         )
 
+        print(f"{r_edge} {f_edge} {edge_t.rel.reverse()}")
```
Comment (jgrillo-grapl) [inline src/python/grapl-model-plugin-deployer/src/grapl_model_plugin_deployer.py:148]: same

File: src/python/grapl-model-plugin-deployer/src/grapl_model_plugin_deployer.py [hunk 4]
```diff
@@ -222,21 +241,40 @@ def provision_schemas(master_graph_client, raw_schemas):
     print(f"Schemas: {schemas}")
     LOGGER.info(f"deploying schemas: {[s.self_type() for s in schemas]}")
 
+    # loaded_builtins = [s for s in schemas if s.self_type() in builtin_schemas]
+    # plugin_schemas = [s for s in schemas if s.self_type() not in builtin_schemas]
+
+    for edges in (s.edges for s in schemas):
+        for f_edge, (edge_t, r_edge) in edges.items():
+            print("pre-reverse", f_edge, edge_t.__dict__, r_edge)
```
Comment (jgrillo-grapl) [inline src/python/grapl-model-plugin-deployer/src/grapl_model_plugin_deployer.py:249]: same

File: src/python/grapl-model-plugin-deployer/src/grapl_model_plugin_deployer.py [hunk 5]
```diff
@@ -222,21 +241,40 @@ def provision_schemas(master_graph_client, raw_schemas):
...
+
     LOGGER.info("init_reverse")
     for schema in schemas:
         schema.init_reverse()
 
+    for edges in (s.edges for s in schemas):
+        for f_edge, (edge_t, r_edge) in edges.items():
+            print("pre-store", f_edge, edge_t.__dict__, r_edge)
```
Comment (jgrillo-grapl) [inline src/python/grapl-model-plugin-deployer/src/grapl_model_plugin_deployer.py:257]: same

File: src/python/grapl-model-plugin-deployer/src/grapl_model_plugin_deployer.py [hunk 6]
```diff
@@ -222,21 +241,40 @@ def provision_schemas(master_graph_client, raw_schemas):
...
     LOGGER.info("Merge the schemas with what exists in the graph")
     dynamodb = get_dynamodb_client()
     for schema in schemas:
         store_schema(dynamodb, schema)
 
-    for schema in schemas:
-        extend_schema(dynamodb, master_graph_client, schema)
+    for edges in (s.edges for s in schemas):
+        for f_edge, (edge_t, r_edge) in edges.items():
+            print("pre-extend", f_edge, edge_t.__dict__, r_edge)
```
Comment (jgrillo-grapl) [inline src/python/grapl-model-plugin-deployer/src/grapl_model_plugin_deployer.py:266]: same

File: src/python/grapl-model-plugin-deployer/src/grapl_model_plugin_deployer.py [hunk 7]
```diff
@@ -222,21 +241,40 @@ def provision_schemas(master_graph_client, raw_schemas):
...
 
     LOGGER.info("Reprovision the graph")
     provision_master_graph(master_graph_client, schemas)
 
+    for schema in schemas:
+        extend_schema(dynamodb, master_graph_client, schema)
+
+    for edges in (s.edges for s in schemas):
+        for f_edge, (edge_t, r_edge) in edges.items():
+            print("2pre-store", f_edge, edge_t.__dict__, r_edge)
```
Comment (jgrillo-grapl) [inline src/python/grapl-model-plugin-deployer/src/grapl_model_plugin_deployer.py:276]: same

File: src/python/grapl-model-plugin-deployer/src/grapl_model_plugin_deployer.py [hunk 8]
```diff
@@ -254,14 +292,18 @@ def query_dgraph_predicate(client: "GraphClient", predicate_name: str):
...
 
-def meta_into_edge(dynamodb, schema, f_edge):
+def meta_into_edge(dynamodb, schema: "Schema", f_edge):
     table = dynamodb.Table(os.environ["BUCKET_PREFIX"] + "-grapl_schema_table")
     edge_res = table.get_item(Key={"f_edge": f_edge})["Item"]
-    print(edge_res)
-    return EdgeT(type(schema), BaseSchema, EdgeRelationship(edge_res["relationship"]))
+    edge_t = schema.edges[f_edge][0]  # type: EdgeT
+
+    print(
```
Comment (jgrillo-grapl) [inline src/python/grapl-model-plugin-deployer/src/grapl_model_plugin_deployer.py:300]: same

File: src/python/grapl-model-plugin-deployer/src/grapl_model_plugin_deployer.py [hunk 9]
```diff
@@ -313,15 +355,27 @@ def query_dgraph_type(client: "GraphClient", type_name: str):
...
+def get_reverse_edge(dynamodb, schema, f_edge):
+    table = dynamodb.Table(os.environ["BUCKET_PREFIX"] + "-grapl_schema_table")
+    edge_res = table.get_item(Key={"f_edge": f_edge})["Item"]
+    return edge_res["r_edge"]
+
+
 def extend_schema(dynamodb, graph_client: GraphClient, schema: "BaseSchema"):
     predicate_metas = query_dgraph_type(graph_client, schema.self_type())
-
+    print(f"predicate_metas, {schema.self_type()}", predicate_metas)
```
Comment (jgrillo-grapl) [inline src/python/grapl-model-plugin-deployer/src/grapl_model_plugin_deployer.py:366]: same

File: src/python/grapl-model-plugin-deployer/src/grapl_model_plugin_deployer.py [hunk 10]
```diff
@@ -313,15 +355,27 @@ def query_dgraph_type(client: "GraphClient", type_name: str):
...
     for predicate_meta in predicate_metas:
         predicate = meta_into_predicate(dynamodb, schema, predicate_meta)
         if isinstance(predicate, PropType):
             schema.add_property(predicate_meta["predicate"], predicate)
         else:
-            schema.add_edge(predicate_meta["predicate"], predicate, "")
+            try:
+                r_edge = get_reverse_edge(dynamodb, schema, predicate_meta["predicate"])
+                print(f'f_edge {predicate_meta["predicate"]} r_edge {r_edge}')
```
Comment (jgrillo-grapl) [inline src/python/grapl-model-plugin-deployer/src/grapl_model_plugin_deployer.py:374]: same

File: src/python/grapl-model-plugin-deployer/src/grapl_model_plugin_deployer.py [hunk 11]
```diff
@@ -313,15 +355,27 @@ def query_dgraph_type(client: "GraphClient", type_name: str):
...
+                schema.add_edge(predicate_meta["predicate"], predicate, r_edge)
+            except Exception as e:
+                print("error on add_edge", e)
```
Comment (jgrillo-grapl) [inline src/python/grapl-model-plugin-deployer/src/grapl_model_plugin_deployer.py:377]: same

**Review by jgrillo-grapl — CHANGES_REQUESTED** [review]
Lots of `print()` calls that should probably be `LOGGER.x` calls instead.

File: src/python/engagement_edge/src/engagement_edge.py [hunk 12]
```diff
@@ -382,6 +389,7 @@ def respond(err, res=None, headers=None):
     else:
         LOGGER.info("Origin did not match")
         # allow_origin = override or ORIGIN
+        # todo: Fixme
```
Comment (inickles-grapl) [inline src/python/engagement_edge/src/engagement_edge.py:392]: is this a todo for this PR or no?

**Review by jgrillo-grapl — APPROVED** [review]

File: src/python/grapl-model-plugin-deployer/src/grapl_model_plugin_deployer.py (hunk 2 above)
Comment (jgrillo-grapl) [reply src/python/grapl-model-plugin-deployer/src/grapl_model_plugin_deployer.py:139]: still

File: src/python/grapl-model-plugin-deployer/src/grapl_model_plugin_deployer.py (hunk 3 above)
Comment (jgrillo-grapl) [reply src/python/grapl-model-plugin-deployer/src/grapl_model_plugin_deployer.py:148]: still

File: src/python/grapl-model-plugin-deployer/src/grapl_model_plugin_deployer.py (hunk 8 above)
Comment (jgrillo-grapl) [reply src/python/grapl-model-plugin-deployer/src/grapl_model_plugin_deployer.py:264]: still

File: src/python/grapl-model-plugin-deployer/src/grapl_model_plugin_deployer.py (hunk 9 above)
Comment (jgrillo-grapl) [reply src/python/grapl-model-plugin-deployer/src/grapl_model_plugin_deployer.py:330]: still

File: src/python/grapl-model-plugin-deployer/src/grapl_model_plugin_deployer.py (hunk 10 above)
Comment (jgrillo-grapl) [reply src/python/grapl-model-plugin-deployer/src/grapl_model_plugin_deployer.py:338]: still

File: src/python/grapl-model-plugin-deployer/src/grapl_model_plugin_deployer.py (hunk 11 above)
Comment (jgrillo-grapl) [reply src/python/grapl-model-plugin-deployer/src/grapl_model_plugin_deployer.py:341]: still

**Review by jgrillo-grapl — CHANGES_REQUESTED** [review]

**Review by jgrillo-grapl — APPROVED** [review]
👍

**Pull request merged**